component/file_cache/file_cache_test.go

```diff
@@ -995,8 +995,14 @@ func (suite *fileCacheTestSuite) TestRenameFileNotInCache() {
 	handle, _ := suite.fileCache.CreateFile(internal.CreateFileOptions{Name: src, Mode: 0777})
 	suite.fileCache.CloseFile(internal.CloseFileOptions{Handle: handle})
 
+	_, err := os.Stat(suite.cache_path + "/" + src)
+	for !os.IsNotExist(err) {
```

Comment:
my only worry for this approach is if we run into a file cache bug where the file doesn't get deleted for one reason or another down the road. So maybe we should break after a given amount of time has passed and if we broke then the test fails

Reply:
Nice suggestion. Will see if we can break it after X retries.

Reply:
corrected on all instances of notExists check